src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,98 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int currentCapacity;
+    private T[] entries;
+    private int size;
+
+    public ArrayList() {
+        entries = (T[]) new Object[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == currentCapacity) {
+            grow();
+        }
+        if (index != size) {
+            checkIndex(index);
+            System.arraycopy(entries, index, entries, index + 1, size - index);
+        }
+        entries[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (list == null) {
+            throw new RuntimeException("List can't be null!");
+        }
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return entries[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkIndex(index);
+        entries[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        T foundObject = entries[index];
+        System.arraycopy(entries, index + 1, entries, index,--size - index);
+        return foundObject;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(entries[i], element)) {
```

Comment:
Dont use Objects class. Read checklist more attendently.

Reply:
Damn... I thought we've already passed that, ok)